### PR TITLE
RSDK-9369: Check for rpc server existence before accessing.

### PR DESCRIPTION
**Pull request by dgottlieb**
I'm kind of curious why a test hasn't hit this. In case you have any ideas. I might dig deeper into that.

### CODE REVIEW

**Comment by benjirewis** [comment]
> I'm kind of curious why a test hasn't hit this. In case you have any ideas. 

Tests must always have an `rpcServer` on the web service or FTDC isn't calling `Stats` on the web service during tests? Not sure.

File: robot/web/web_c.go [hunk 1]
```diff
@@ -57,10 +57,21 @@ type webService struct {
...
 
+type stats struct {
+	RPCServer any
+}
+
 func (svc *webService) Stats() any {
-	return struct {
-		RPCServer any
-	}{svc.rpcServer.Stats()}
+	if haveLock := svc.mu.TryLock(); !haveLock {
```
Comment (benjirewis) [inline robot/web/web_c.go:65]: Hmm `TryLock`. You avoiding a long lock here when the web service is reconfiguring? Or worried/aware of some kind of deadlock?

Comment (dgottlieb) [reply robot/web/web_c.go:65]: You're right, this needs to be justified in words

Comment (dgottlieb) [reply robot/web/web_c.go:65]: Oops, also this PR is wrong -- that whole changing the schema thing.

File: robot/impl/local_robot.go [hunk 2]
```diff
@@ -244,7 +245,14 @@ func (r *localRobot) Logger() logging.Logger {
 
 // StartWeb starts the web server, will return an error if server is already up.
 func (r *localRobot) StartWeb(ctx context.Context, o weboptions.Options) (err error) {
-	return r.webSvc.Start(ctx, o)
+	time.Sleep(time.Second)
```
Comment (benjirewis) [inline robot/impl/local_robot.go:248]: What's up with this sleep?

**Review by benjirewis — APPROVED** [review]
LGTM thanks!

**Pull request merged**